Load all of the necessary parts of the patient bundle into the new UI

jtquach1 opened the pull request:
# WIP

Changes made:
- Preloaded Cancer Type, Cancer Subtype, Metastasis, Stage, ECOG Score, Karnofsky Score, Biomarkers, Radiation, Surgery, Medications

To-do:
- Update to use SNOMED/LOINC codes
- Test each form entry with patient bundles

noranda commented:
@jtquach1 I refactored some of the FHIR conversion utils, but otherwise this looks great. We should add the additional codes and some tests as you noted.

noranda reviewed (changes requested):
Add additional codes and tests.

jtquach1 commented:
> Add additional codes and tests.

I added tests for `src/utils/fhirConversionUtils.ts` as well as multiple helper functions (not sure if I should test those as well) and mock bundles based on some sample Moonshot dev patients and the sample mCODE patients from #6 . I also changed the metastasis field to an autocomplete through a chat with Zach as it's possible that a patient could have multiple secondary cancer conditions (either metastasized from a primary cancer condition or from another secondary cancer condition). 

I did use `@types/fhir/r4` in the refactoring since it's more informative on some resources, but I don't think it  interferes with `@types/fhirclient`; I tried making the sample bundles based on what's in the mCODE and core FHIR IGs. 

I also created maps for SNOMED and LOINC codes and and refactored the Karnofsky/ECOG functions to use them. So far they're hardcoded but maybe we could use the SNOMED/LOINC API in the future?

dmpotter44 commented on src/utils/fhirConversionUtils.ts:
```diff
@@ -0,0 +1,207 @@
+export const convertFhirPrimaryCancerCondition = (bundle: fhirclient.FHIR.Bundle): PrimaryCancerCondition => {
+  const condition = bundle?.entry?.[0]?.resource as Condition;
+  return {
+    cancerType: condition?.bodySite?.[0]?.coding?.[0]?.display ?? null,
+    cancerSubtype: condition?.code?.coding?.[0]?.display ?? null,
```
The cancer type should be the `Condition.code`. The `Condition.bodySite` code, if present, would indicate where the condition is located. From the FHIR R4 spec on the `bodySite` property: "Only used if not implicit in code found in Condition.code."

From the MCode spec it looks like the exact cancer type is a combination of the `Condition.code` and `Condition.bodySite` codes. However, most of the codes we have seem to imply a body site. For example, for "Breast Cancer," we have the SNOMED code "372137005 - Primary malignant neoplasm of breast (disorder)."

I don't know exactly what the subtype should be here. The example I have uses the [MCode histology morphology behavior extension](https://hl7.org/fhir/us/mcode/StructureDefinition-mcode-histology-morphology-behavior.html) which I guess could be a "subtype"?

So fixing this isn't quite as simple as using the `Condition.code` for the type and the `Condition.bodySite` for the subtype. It may involve some more clarification on what these fields are.

dmpotter44 reviewed (changes requested):
We need to figure out exactly what the cancer type and cancer subtype fields are. I'm fairly sure that the current code is using the wrong fields, but unfortunately I don't know exactly what the right fields are.

noranda commented on src/pages/search.tsx:
```diff
@@ -49,22 +90,65 @@ export const getServerSideProps: GetServerSideProps = async context => {
+    getMedicationStatement('mcode-cancer-related-medication-statement'),
   ]);
 
+  // console.log('fhirTumorMarkers.entry', JSON.stringify(fhirTumorMarkers.entry));
```
Lets remove these console logs.

noranda reviewed (changes requested):
This looks good! I would remove the commented out console logs and we'll need to address @dmpotter44 's comment, but otherwise this looks good to me. Nice work!
